apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx

```diff
@@ -46,185 +39,209 @@ interface CustomEventDialogProps {
     scheduleNames: string[];
 }
 
-interface CustomEventDialogState extends RepeatingCustomEvent {
-    open: boolean;
-    scheduleIndices: number[];
-}
-
-const defaultCustomEvent: RepeatingCustomEvent = {
+const defaultCustomEventValues: RepeatingCustomEvent = {
     start: '10:30',
     end: '15:30',
     title: '',
     days: [false, false, false, false, false, false, false],
     customEventID: 0,
 };
 
-class CustomEventDialog extends PureComponent<CustomEventDialogProps, CustomEventDialogState> {
-    state: CustomEventDialogState = {
-        open: false,
-        ...(this.props.customEvent || defaultCustomEvent),
-        scheduleIndices: [],
-    };
+function CustomEventDialogs(props: CustomEventDialogProps) {
+    const [skeletonMode, setSkeletonMode] = useState(AppStore.getSkeletonMode());
+
+    const [open, setOpen] = useState(false);
+    const [scheduleIndices, setScheduleIndices] = useState<number[]>([]);
+    const [start, setStart] = useState(defaultCustomEventValues.start);
+    const [end, setEnd] = useState(defaultCustomEventValues.end);
+    const [title, setTitle] = useState(defaultCustomEventValues.title);
+    const [days, setDays] = useState(defaultCustomEventValues.days);
 
-    handleOpen = () => {
-        this.setState({ open: true, scheduleIndices: [AppStore.schedule.getCurrentScheduleIndex()] });
+    useEffect(() => {
+        const handleSkeletonModeChange = () => {
+            setSkeletonMode(AppStore.getSkeletonMode());
+        };
+
+        AppStore.on('skeletonModeChange', handleSkeletonModeChange);
+
+        return () => {
+            AppStore.off('skeletonModeChange', handleSkeletonModeChange);
+        };
+    }, []);
+
+    const handleOpen = useCallback(() => {
+        setOpen(true);
+        setScheduleIndices([AppStore.schedule.getCurrentScheduleIndex()]);
 
         logAnalytics({
             category: analyticsEnum.calendar.title,
             action: analyticsEnum.calendar.actions.CLICK_CUSTOM_EVENT,
         });
-    };
+    }, [setOpen]);
 
-    handleClose = (cancel: boolean) => {
-        if (!cancel) {
-            logAnalytics({
-                category: analyticsEnum.calendar.title,
-                action: analyticsEnum.calendar.actions.ADD_CUSTOM_EVENT,
-            });
-            if (this.props.onDialogClose) this.props.onDialogClose();
-            this.handleAddToCalendar();
-        }
-
-        this.setState({
-            ...this.state,
-            open: false,
-        });
-    };
-
-    handleEventNameChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
-        this.setState({ title: event.target.value });
-    };
+    const handleClose = useCallback(() => {
+        setOpen(false);
+    }, [setOpen]);
 
-    handleEndTimeChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
-        this.setState({ end: event.target.value });
-    };
+    const handleSubmit = () => {
```

Comment:
This is a new function. Previously `handleClose` had a conditional inside of it telling it whether or not to add the custom event to calendar.

That doesn't make a ton of sense as `handleClose` should close the component, so I've made a new function for this purpose.